glom/core.py

```diff
@@ -1460,6 +1486,96 @@ def _format_t(path, root=T):
     return "".join(prepr)
 
 
+class Val(object):
+    """Val objects are specs which evaluate to the wrapped value.
+
+    >>> target = {'a': {'b': 'c'}}
+    >>> spec = {'a': 'a.b', 'readability': Val('counts')}
+    >>> pprint(glom(target, spec))
+    {'a': 'c', 'readability': 'counts'}
+
+    Instead of accessing ``'counts'`` as a key like it did with
+    ``'a.b'``, :func:`~glom.glom` just unwrapped the Val and
+    included the value.
+
+    :class:`~glom.Val` takes one argument, the value to be returned.
+    """
+    def __init__(self, val):
+        self.val = val
+
+    def glomit(self, target, scope):
+        return self.val
+
+    def __repr__(self):
+        cn = self.__class__.__name__
+        return '%s(%s)' % (cn, bbrepr(self.val))
+
+
+class _Vars(object):
+    """
+    This is the run-time partner of :class:`Vars` -- this is what
+    actually lives in the scope and stores run-time values.
+    """
+    def __init__(self, base, defaults):
+        self.__dict__ = dict(base)
+        self.__dict__.update(defaults)
+
+    def __repr__(self):
+        return "Vars(%s)" % bbrepr(self.__dict__)
+
+
+class Vars(object):
+    """
+    :class:`Vars` is a helper that can be used with `Let` in order to
+    store shared mutable state.
+
+    Takes the same arguments as `dict()`.
+
+    Arguments here should be thought of the same way as default arguments
+    to a function.  Each time the spec is evaluated, the same arguments
+    will be referenced; so, think carefully about mutable data structures.
+    """
+    def __init__(self, base=(), **kw):
+        dict(base)  # ensure it is a dict-compatible first arg
+        self.base = base
+        self.defaults = kw
```

Comment:
why not combine these here instead of in the glomit?

Reply:
for the repr

Reply:
(I'm not super attached to the behavior -- just explaining that was the only reason I did it this way,

`Vars(a=1, {'b': 2})` would repr correctly

it doesn't seem so important if everything is a string, but if there are rando objects in there being used as keys, it seems that it might look really weird `Vars(a=1, 2=3)`

mostly my expectation is 99% of the time people will use kwargs)